ros2service/ros2service/verb/call.py

```diff
@@ -44,23 +43,25 @@ def add_arguments(self, parser, cli_name):
             help='Values to fill the service request with in YAML format ' +
                  '(e.g. "{a: 1, b: 2}"), ' +
                  'otherwise the service request will be published with default values')
-
-        def float_or_int(input_string):
-            try:
-                value = float(input_string)
-            except ValueError:
-                raise argparse.ArgumentTypeError("'%s' is not an integer or float" % input_string)
-            return value
-
         parser.add_argument(
-            '-r', '--repeat', metavar='N', type=float_or_int,
-            help='Repeat the call every N seconds')
+            '-r', '--rate', metavar='N', type=float, default=1,
+            help='Calling rate (hz) default to 1')
+        parser.add_argument(
+            '-1', '--once', action='store_true',
+            help='Call service once and exit')
 
     def main(self, *, args):
-        return requester(args.service_type, args.service_name, args.values, args.repeat)
+        if args.rate <= 0:
+            raise ValueError('rate must be greater than zero')
+
+        return requester(args.service_type, args.service_name, args.values, args.rate, args.once)
 
 
-def requester(service_type, service_name, values, repeat):
+def requester(service_type, service_name, values, rate, once):
+    if rate is not None:
+        period = 1. / rate
+    else:
```

Comment:
This is now never the case.

Reply:
forgot to commit local changes :/
https://github.com/ros2/ros2cli/pull/66/commits/29015d6be6193997f8d9179322e4f4957e49e05d